tests/integration/istioio/examples/scripts/bookinfo.txt

```diff
@@ -0,0 +1,128 @@
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -u
+set -o pipefail
+
+REPEAT=${REPEAT:-100}
+THRESHOLD=${THRESHOLD:-20}
+
+function verify_services {
+  unmatchedText=""
+  wantTextPattern=("$@")
+  goodResponse=0
+
+  for ((i=1; i<="$REPEAT"; i++)); do
+    tmpfile=$(mktemp /tmp/verify-services.XXXXXX)
+    set +e
+    $(kubectl get services > "${tmpfile}")
+    set -e
+    body=$(cat ${tmpfile})
+
+    matchedText=0
+    for want in "${wantTextPattern[@]}"; do
+       if [ "`grep "${want}" "${tmpfile}"`" != "" ]; then
+        matchedText=$((matchedText + 1))
+      else
+        unmatchedText="${want}"
+      fi
+    done
+
+    if [[ "$matchedText" == "$#" ]]; then
+      goodResponse=$((goodResponse + 1))
+    else
+      goodResponse=0
+    fi
+
+    if (( "$goodResponse">="$THRESHOLD" )); then
+      return 0
+    fi
+  done
+  rm "$tmpfile"
+
+  echo -e "did not match text: ${unmatchedText} in ${body}\n"
+  return 1
+}
+
+function verify_pods {
+  unmatchedText=""
+  wantTextPattern=("$@")
+  goodResponse=0
+
+  for ((i=1; i<="$REPEAT"; i++)); do
+    tmpfile=$(mktemp /tmp/verify-pods.XXXXXX)
+    set +e
+    $(kubectl get pods > "${tmpfile}")
+    set -e
+    body=$(cat ${tmpfile})
+
+    matchedText=0
+    for want in "${wantTextPattern[@]}"; do
+       if [ "`grep "${want}" "${tmpfile}"`" != "" ]; then
+        matchedText=$((matchedText + 1))
+      else
+        unmatchedText="${want}"
+      fi
+    done
+
+    if [[ "$matchedText" == "$#" ]]; then
+      goodResponse=$((goodResponse + 1))
+    else
+      goodResponse=0
+    fi
+
+    if (( "$goodResponse">="$THRESHOLD" )); then
+      return 0
+    fi
+  done
+  rm "$tmpfile"
+
+  echo -e "did not match text: ${unmatchedText} in ${body}\n"
+  return 1
+}
+
+# Remove the injection label to prevent following command to fail
+kubectl label namespace default istio-injection-
+
+# $snippet automatic_sidecar_injection.sh syntax="bash"
+$ kubectl label namespace default istio-injection=enabled
+# $endsnippet
+
+# $snippet deploy_bookinfo_application.sh syntax="bash"
+$ kubectl apply -f @samples/bookinfo/platform/kube/bookinfo.yaml@
+# $endsnippet
+
+# $snippet verify_service.sh syntax="bash" outputis="text" outputsnippet="true"
+$ kubectl get services
+# $endsnippet
+
+verify_services "NAME *TYPE *CLUSTER-IP *EXTERNAL-IP *PORT(S) *AGE" "details *ClusterIP *[0-9.]* *.none. *9080/TCP" "kubernetes *ClusterIP *[0-9.]* *.none. *443/TCP" "productpage *ClusterIP *[0-9.]* *.none. *9080/TCP" "ratings *ClusterIP *[0-9.]* *.none. *9080/TCP" "reviews *ClusterIP *[0-9.]* *.none. *9080/TCP"
```

Comment:
Is this how we are running the tests now? I thought just the user facing snippets were in bash and the rest was in golang using the existing mechanisms we have for waiting for pods to be deployed, etc. Won't we have to copy this for every test?

Reply:
I think it is just bookinfo, unless other tests verify the output.
This pattern is emerging from the security test. I'm not sure if the best approach is to use TokenVerifier with Regex in go integration test code

Reply:
Yep. Go is still an option, but the majority of the test is now written in these scripts. More closely models the format that docs writers are using. 

Waiting for pods can be done as follows: 

https://github.com/istio/istio/blob/master/tests/integration/istioio/security/mtls_migration_test.go#L55..L57

Reply:
@fpesce I'd like to simplify the arguments to this function so we can standardize around it.

Ideally, I think it would be nice to just have the call site look something like this:

```bash
wait_for_pods "my-namespace"
```

WDYT?

Reply:
@fpesce so you're just verifying that the output of `kubectl get services` is correct? We should probably just use the snippet verification for this: https://github.com/istio/istio/blob/master/pkg/test/istioio/script.go#L96